### ISSUE TITLE
Crash in implicitclass: Assertion `(old_top == initial_top (av) && old_size == 0) || ((unsigned long) (old_size) >= MINSIZE && prev_inuse (old_top) && ((unsigned long) old_end & (pagesize - 1)) == 0)\' failed.

### ISSUE THREAD

**Issue by paulmenzel** [OpenPrinting/cups-filters#162]
With cups-filters 1.25.6 printing fails with the error below.

```
D [01/Oct/2019:11:15:54 +0200] [Job 168] Started backend /usr/lib/cups/backend/implicitclass (PID 14124)
D [01/Oct/2019:11:15:54 +0200] [Job 168] Read cups-browsed-dest-printer option: \"168 ipps://EPSON73FFDE.local:631/ipp/print raster 600dpi\"
D [01/Oct/2019:11:15:54 +0200] [Job 168] Received destination host name from cups-browsed: printer-uri ipps://EPSON73FFDE.local:631/ipp/print raster 600dpi
D [01/Oct/2019:11:15:54 +0200] [Job 168] Received job for the printer with the destination uri - ipps://EPSON73FFDE.local:631/ipp/print, Final-document format for the printer - raster and requested resolution - 600dpi
D [01/Oct/2019:11:15:54 +0200] [Job 168] Debug: implicitclass - copying to temp print file \"/var/spool/cups/tmp/0372c5d96a359\"
D [01/Oct/2019:11:15:54 +0200] [Job 168] pdftopdf: Last filter determined by the PPD: -; FINAL_CONTENT_TYPE: application/pdf => pdftopdf will log pages in page_log.
D [01/Oct/2019:11:15:54 +0200] [Job 168] PDF interactive form and annotation flattening done via QPDF
D [01/Oct/2019:11:15:54 +0200] [Job 168] PAGE: 1 1
D [01/Oct/2019:11:15:54 +0200] [Job 168] Removing cups-browsed-dest-printer option from argumentsmalloc.c:2389: sysmalloc: Assertion `(old_top == initial_top (av) && old_size == 0) || ((unsigned long) (old_size) >= MINSIZE && prev_inuse (old_top) && ((unsigned long) old_end & (pagesize - 1)) == 0)\' failed.
D [01/Oct/2019:11:15:54 +0200] [Job 168] PID 14123 (/usr/lib/cups/filter/pdftopdf) exited with no errors.
D [01/Oct/2019:11:15:54 +0200] [Job 168] PID 14124 (/usr/lib/cups/backend/implicitclass) crashed on signal 6.
D [01/Oct/2019:11:15:54 +0200] [Job 168] Hint: Try setting the LogLevel to "debug" to find out more.
D [01/Oct/2019:11:15:54 +0200] [Job 168] Backend returned status -134 (crashed)
```

Report in Ubuntu: https://bugs.launchpad.net/ubuntu/+source/cups-filters/+bug/1845548

**Comment by paulmenzel** [comment]
Downgrading *cups-browsed*, *cups-filters*, *libcupsfilters1*, and *cups-filters-core-drivers* to version 1.25.5 works around the issue.

Only downgrading *cups-browsed* (and *cups-filters*) resolves the crash, but printing still does not work.

    File \'\' not found
    Unable to add document to print job.

**Comment by deepak0405** [comment]
Hey @paulmenzel 
I tried reproducing the error on my system but was unable to do so. I ran the latest `1.25.6` cups-filters package and tested a job on ippserver, but didn't get this error.

I have found a few code lines which could have been the reason for the bug. I need your help to see whether these changes solve the bug. Can you please try running the cups-filters from the branch https://github.com/deepak0405/cups-filters/tree/solve-assertion-error  ?

**Comment by tillkamppeter** [comment]
@paulmenzel, If you use Ubuntu, I have packaged @dheeraj135, branch and put into my PPA:
https://launchpad.net/~till-kamppeter/+archive/ubuntu/ppa/+packages
cups-filters_1.25.6+git20191002-0ubuntu1~ppa1

**Comment by paulmenzel** [comment]
I built the package under Debian Sid/unstable with the commit from https://github.com/deepak0405/cups-filters/tree/solve-assertion-error. Printing worked. The error message `File \'\' not found` is still there, but it might be Evince related, and I filed an [issue upstream](https://gitlab.gnome.org/GNOME/evince/issues/1277).

**Comment by tillkamppeter** [comment]
This is exactly what I have released as 1.25.7 and uploaded for Ubuntu Eoan.
Thank you very much for the feedback.
